### PR TITLE
Email verification on profile

### LINKED ISSUE #10378: ✨ Profile changes for email verification
## ✨ Feature

We want to clearly show to users if their emails are validated or not. If they are not, we provide a link on the profile for them to validate. We are also adding the new "work email" field to the profile in the "employee info" section. Rename department to "home department".

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

Include a link to the design file (if it exists).

### 🧑‍🎨 Designer

Include the author of the linked design file's name (or even better, their `@githubUsername`) for future reference.

### Verified state
image

### Unverified state
image

### Email verification

There are 3 different flows that reach a similar email verification page.

1. Verify email on profile
2. Registration takes you to email verification as well
3. And verify email on profile step of the application

The heading and breadcrumb should match the ones in the flow that's being used.

image

## 🙋‍♀️ Proposed Implementation

**(Questions)**

In the profile flow, the email verification page should be:
https://talent.canada.ca/en/applicant/email-verification
(or /contact-email-verification if dev prefers or /applicant/personal-information/... if that makes more sense with breadcrumb)

Breadcrumb should be:
Home > Profile and applications > Personal information > Verify email

The H1 should follow the profile H1, like here:
image

Once complete, user should be brought back to Personal information page.

Is everything else covered by the reusable component #10687 ?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] New chip (pill?) for verification beside personal email
- [ ] 2 states for the chip, "verified" or "unverified"
- [ ] Verify link provided beside the "unverified" chip
- [ ] Verify link brings user to a new page for verifying email and code is sent by email (should be triggered by the component, I think)
- [ ] Editing the contact email causes it to become "unverified" again, but only if the email actually changes. Hitting "edit this section" changing something else (eg. Province or territory) should not "unverify" the personal email.
- [ ] Verifying email brings user back to personal information page with email verified
- [ ] Tests?
- [ ] url `https://talent.canada.ca/en/applicant/verify-contact-email`

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] #10687
- [ ] #10822

```

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #10378 

## 👋 Introduction

This branch allows users to verify their contact email address.

## 🕵️ Details

I reworked the grid of the display items a bit.  The previous column-based layout stacked up strangely in the mobile view - for example, splitting the two names and the three languages.  I think the updated layout is more what was intended.
image

### :crossed_fingers: Cheater

1. Log in as a user
2. Navigate to the user profile
3. Click the verify email link
4. Look in the jobs table and look in the payload field to find the code that would be sent
5. Enter the wrong code in the box, observe the error
6. Enter the right code in the box, observer the verification

### :burrito: The full enchilada

1. Set up .env with GCNotify
2. Start the queue workers
3. Log in as a user
4. Change email to a whitelisted email
3. Click the verify email link
5. Wait for the email to arrive
6. Enter the wrong code in the box, observe the error
7. Enter the right code in the box, observer the verification

### :notebook: Other

- [ ] Profile grid looks good wide and narrow viewports
- [ ] Can send a new email from the "send again" button on the verification form
- [ ] Verification doesn't appear in application profile page

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx
Comment (vd1992) [inline apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:1]: Tiny space present slightly throwing off alignment

image

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx [hunk 1]
```diff
@@ -36,128 +53,197 @@ const Display = ({
...
   const notProvided = intl.formatMessage(commonMessages.notProvided);
+  const { userAuthInfo } = useAuthorization();
+  const navigate = useNavigate();
+  const routes = useRoutes();
+  const applicationContext = useApplicationContext();
 
-  return (
+  // only show email verification if NOT in an application currently
+  const showEmailVerification = !applicationContext.currentStepOrdinal;
+
```
Comment (vd1992) [inline apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: Confused, the issue requests that it **_IS_** present in that context

image

Comment (vd1992) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: Registration does not appear to connect there either

File: api/app/GraphQL/Mutations/PoolCandidateSnapshot.graphql
Comment (vd1992) [inline api/app/GraphQL/Mutations/PoolCandidateSnapshot.graphql:1]: Think you need to double check PHP tests due to this change

**Review by vd1992 — CHANGES_REQUESTED** [review]
Thanks for the lite version of the instructions, made this a smoother experience 😆 
Please take a look at the below

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx (hunk 1 above)
Comment (petertgiles) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: @gobyrne could you please confirm the scope of this issue?  Just profile or also application and create-account?

Comment (petertgiles) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: I figured the other two were here:
image

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx
Comment (petertgiles) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:1]: Good catch!
[fix margin](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10893/commits/94bb4b1c61a0d231ce4c9a62c26f79e3ec1e4a9c)
image

File: api/app/GraphQL/Mutations/PoolCandidateSnapshot.graphql
Comment (petertgiles) [reply api/app/GraphQL/Mutations/PoolCandidateSnapshot.graphql:1]: Thanks.  I think this fixes it.
[add field to snapshot](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10893/commits/f7320e19f707e2032a93aadd3b3bf14b8595d8bc)

**Comment by gobyrne** [comment]
This issue is only for #1 on the list Vachan shared. There are separate issues for 2 and 3.

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx (hunk 1 above)
Comment (vd1992) [inline apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: Oh, another case where it is (probably) best to hide it. Viewing an application
Clicking verify triggers verification of the logged in admin 

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Looks solid overall, just the one tiny but important thing left

File: apps/web/src/components/Profile/components/PersonalInformation/Display.tsx (hunk 1 above)
Comment (vd1992) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: Addressed below

Comment (vd1992) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx:63]: Might be best to take a whitelist approach to be safest.
So something like 

`const showEmailVerification = isPersonalInformationContext;`

And then additional allowed places can be tagged in like 

`|| isApplicationContext`

Comment (petertgiles) [reply apps/web/src/components/Profile/components/PersonalInformation/Display.tsx]: Shoot, good catch.  I forgot about the admin display.  I like your suggestion of switching to default-off.
[switch to whitelist](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10893/commits/1efeac774d9f383361e439f223a081204a1ec4fe)

**Review by vd1992 — APPROVED** [review]
Little bit of lint aside, looks good. Great stuff!

**Pull request merged**